### PR TITLE
[Issues#17472] add some test for SingleTableDropSchemaMetadataValidator

### PR DESCRIPTION
**Pull request by complone**
Fixes #[ISSUSE_17472](https://github.com/apache/shardingsphere/issues/17472).

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                mock(DatabaseType.class), mock(ShardingSphereResource.class, RETURNS_DEEP_STUBS), ruleMetaData, Collections.emptyMap());
+        SingleTableRule singleTableRule = new SingleTableRule(new SingleTableRuleConfiguration(), DefaultDatabase.LOGIC_NAME, mock(DatabaseType.class),
+                createSingleDataSourceMap(), Collections.emptyList(), new ConfigurationProperties(new Properties()));
+        DropSchemaStatement sqlStatement = new PostgreSQLDropSchemaStatement();
+        sqlStatement.getSchemaNames().add(new IdentifierValue("t_order_item"));
+        sqlStatement.setParameterCount(1);
+        SQLStatementContext<DropSchemaStatement> sqlStatementContext = new CommonSQLStatementContext<>(sqlStatement);
+        Optional<SingleTableMetadataValidator> result = SingleTableMetadataValidatorFactory.newInstance(sqlStatementContext.getSqlStatement());
+        String msg = "";
+        try {
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java:64]: @complone May replace this logic with `@Test(expected = ShardingSphereException.class) is better.`

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.postgresql.ddl.PostgreSQLDropSchemaStatement;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.mock;
+
+public final class SingleTableDropSchemaMetadataValidatorTest {
+    
+    @Test
+    public void assertGetInstance() {
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java:51]: `assertValidate` may be better.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by strongduanmu** [comment]
Hi @complone, does this pr has any progress?

**Comment by complone** [comment]
> Hi @complone, does this pr has any progress?

yes. i will fix it today

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        DropSchemaStatement sqlStatement = new PostgreSQLDropSchemaStatement();
+        sqlStatement.getSchemaNames().add(new IdentifierValue("t_order_item"));
+        sqlStatement.setParameterCount(1);
+        SQLStatementContext<DropSchemaStatement> sqlStatementContext = new CommonSQLStatementContext<>(sqlStatement);
+        Optional<SingleTableMetadataValidator> result = SingleTableMetadataValidatorFactory.newInstance(sqlStatementContext.getSqlStatement());
+        result.ifPresent(optional -> optional.validate(singleTableRule, sqlStatementContext, database));
+    }
+    
+    private Map<String, DataSource> createSingleDataSourceMap() {
+        Map<String, DataSource> result = new HashMap<>(2, 1);
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java:67]: There is only `1` value in map, but the initial size is `2`

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/fixtrue/FixtureRule.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.singletable.fixtrue;
+
+import org.apache.shardingsphere.infra.rule.ShardingSphereRule;
+
+public final class FixtureRule implements ShardingSphereRule {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/fixtrue/FixtureRule.java:22]: If no SPI usage here, could you remove the class and just use mock only?

File: shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        ShardingSphereRuleMetaData ruleMetaData = new ShardingSphereRuleMetaData(Collections.emptyList(), Collections.singleton(rule));
+        ShardingSphereDatabase database = new ShardingSphereDatabase(DefaultDatabase.LOGIC_NAME,
+                mock(DatabaseType.class), mock(ShardingSphereResource.class, RETURNS_DEEP_STUBS), ruleMetaData, Collections.emptyMap());
+        SingleTableRule singleTableRule = new SingleTableRule(new SingleTableRuleConfiguration(), DefaultDatabase.LOGIC_NAME,
+                createSingleDataSourceMap(), Collections.emptyList(), new ConfigurationProperties(new Properties()));
+        DropSchemaStatement sqlStatement = new PostgreSQLDropSchemaStatement();
+        sqlStatement.getSchemaNames().add(new IdentifierValue("t_order_item"));
+        sqlStatement.setParameterCount(1);
+        SQLStatementContext<DropSchemaStatement> sqlStatementContext = new CommonSQLStatementContext<>(sqlStatement);
+        Optional<SingleTableMetadataValidator> result = SingleTableMetadataValidatorFactory.newInstance(sqlStatementContext.getSqlStatement());
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-single-table/shardingsphere-single-table-core/src/test/java/org/apache/shardingsphere/singletable/route/validator/ddl/SingleTableDropSchemaMetadataValidatorTest.java:62]: `result` is for return value only, please do not name the variable name as `result` if method return void

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Comment by complone** [comment]
FixtureRule only exists as a construction condition and is not provided as spi, otherwise it will fail to execute successfully in jdk17.

reason:
```
org.apache.shardingsphere.infra.metadata.database.rule.ShardingSphereRuleMetaData cannot be applied to given types;
Error: required: java.util.Collection<org.apache.shardingsphere.infra.rule.ShardingSphereRule>
Error: found: java.util.List<java.lang.Object>,java.util.List<java.lang.Object>
Error: reason: actual and formal argument lists differ in length
```

**Comment by linghengqian** [comment]
Obviously you need to merge the master branch.

**Pull request closed without merging**